File: auth-service/src/main/java/com/petclinic/authservice/security/Roles.java
package com.petclinic.authservice.security;

/**
 * Enum for roles in the application
 *
 * @author Dylan Brassard
 * @since 2023-09-17
 */
public enum Roles {
    ADMIN ( "ADMIN"),
    VET ( "VET"),
    OWNER ( "OWNER"),
    INVENTORY_MANAGER ( "INVENTORY_MANAGER"),
    ANONYMOUS ( "ANONYMOUS"),
    ALL ( "ALL");


    private final String role;

    Roles(String roleName) {
        this.role = roleName;
    }

    public String toString() {
        return this.role;
    }
}

File: employee-frontend/src/app/shared/models/roles.ts
export const Roles = {
  admin: 'ADMIN',
  owner: 'OWNER',
  receptionist: 'RECEPTIONIST',
  vet: 'VET',
  inventoryManager: 'INVENTORY_MANAGER',
} as const;

export type Roles = (typeof Roles)[keyof typeof Roles];

export const CUSTOMER_ROLES: Roles[] = [Roles.owner];
export const EMPLOYEE_ROLES: Roles[] = [
  Roles.admin,
  Roles.vet,
  Roles.receptionist,
  Roles.inventoryManager,
];
